libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwnerFactories.scala

```diff
@@ -61,4 +60,14 @@ trait ResourceOwnerFactories[Context] {
 
   def forTimer(acquire: () => Timer): AbstractResourceOwner[Context, Timer] =
     new TimerResourceOwner(acquire)
+
+  def forResource[T](
```

Comment:
I don't think you need this. If you need to lift a Resource to a ResourceOwner, there is some mismatch in the factories. Can't you change the monad in the `SandboxOnXRunner.run` to `ResourceOwner`? (it would actually look better in there)

Reply:
removed this change and improved as per your suggestion, thank you!